modal/object.py

```diff
@@ -27,7 +28,7 @@ def _get_environment_name(environment_name: Optional[str], resolver: Optional[Re
         return config.get("environment")
 
 
-class _Object:
+class _Object(Generic[O]):
```

Comment:
I am not totally sure that inheriting from `Generic` here makes sense, although it did silence the typing error we were otherwise getting here. I say that because the `O` TypeVar is used to represent specific subclasses, rather than a different type of object that this class can act on. What do you think @freider?